CMakeLists.txt

```diff
@@ -269,10 +269,12 @@ if (OPM_ENABLE_PYTHON)
   endif()
   find_package(pybind11 2.2 CONFIG)
   if (NOT pybind11_FOUND)
-      add_subdirectory(python/pybind11)
-      if (NOT pybind11_INCLUDE_DIRS)
-        set(pybind11_INCLUDE_DIRS ${PYBIND11_INCLUDE_DIR} ${PYTHON_INCLUDE_DIRS})
-      endif()
+      include(FetchContent)
+      FetchContent_Declare(pybind11
+                           URL https://github.com/pybind/pybind11/archive/refs/tags/v2.11.0.tar.gz
+                           URL_HASH SHA512=10edfdc499ed70234de0796f79b9f3a7a14d4baf01fb09fedbd9126e0a13d9cdeb213950c53a8f7a67cd6049b1db31efacfe2192098236b242ad3685967ea907
```

Comment:
Cool, I like this idea. Then we do not have to manually update the source tree with hundreds of file each time we want to upgrade pybind11.

When I tried this with cmake, I ran into this issue https://stackoverflow.com/a/75007891/2173773. I am using cmake version 3.22.1. The issue can be fixed by putting `DOWNLOAD_EXTRACT_TIMESTAMP ON` before the URL and URL_HASH properties (tested on Ubuntu 22.04).

Reply:
moved.